contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/ArbitrumInboxInterface.sol";
+import "../interfaces/AggregatorValidatorInterface.sol";
+import "../interfaces/FlagsInterface.sol";
+import "../SimpleWriteAccessController.sol";
+
+contract ArbitrumValidator is SimpleWriteAccessController, AggregatorValidatorInterface {
+
+  bytes4 constant private RAISE_SELECTOR = FlagsInterface.raiseFlag.selector;
+  bytes4 constant private LOWER_SELECTOR = FlagsInterface.lowerFlags.selector;
+
+  address private s_flagsAddress;
+  // Follows: https://eips.ethereum.org/EIPS/eip-1967
+  address constant private s_arbitrumFlag = address(bytes20(bytes32(uint256(keccak256("chainlink.flags.arbitrum-offline")) - 1)));
+  int256 private s_lastChangedAnswer;
+
+  ArbitrumInboxInterface private s_arbitrumInbox;
+  SimpleWriteAccessController private s_gasConfigAccessController;
+
+  struct GasConfiguration {
+    uint256 maximumSubmissionCost;
+    uint32 maximumGasPrice;
+    uint256 gasCostL2;
+    address refundableAddress;
+  }
+  GasConfiguration private s_gasConfig;
+  uint32 constant private s_L2GasLimit = 30000000;
+
+  /**
+   * @param aggregatorAddress default aggregator with access to validate
+   * @param inboxAddress address of the Arbitrum Inbox L1 contract
+   * @param flagAddress address of the Chainlink L2 Flags contract
+   * @param gasConfigAccessController address of the access controller for managing gas price on Arbitrum
+   * @param maxSubmissionCost maximum cost willing to pay on L2
+   * @param maximumGasPrice maximum gas price to pay on L2
+   * @param gasCostL2 value to send to L2 to cover gas fee
+   * @param refundableAddress address where gas excess on L2 will be sent
+   */
+  constructor(
+    address aggregatorAddress,
+    address inboxAddress,
+    address flagAddress,
+    address gasConfigAccessController,
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  ) {
+    require(flagAddress != address(0), "Invalid Flags contract address");
+    s_arbitrumInbox = ArbitrumInboxInterface(inboxAddress);
+    s_gasConfigAccessController = SimpleWriteAccessController(gasConfigAccessController);
+    s_flagsAddress = flagAddress;
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+
+    this.addAccess(aggregatorAddress);
```

Comment:
Could we give default access to the `aggregatorAddress`? Apparently this won't work

Reply:
@RodrigoAD why wouldn't this work?

Reply:
This should work, but nvm, it's ok not to mention the aggregator and just add access externally.